modules/markup_field/src/Plugin/Field/FieldType/MarkupFieldItem.php

```diff
@@ -0,0 +1,290 @@
+    $element['field'] = [
+      '#type' => 'select',
+      '#title' => t('Field to be rendered'),
+      '#options' => $options,
+      '#default_value' => $this->getSetting('field'),
+    ];
+    return $element;
+  }
+
+  /**
+   * Get assets.
+   *
+   * @param array $libraries
+   *   List of libraries.
+   *
+   * @return array
+   *   Assets build.
+   */
+  public function getAssetsFromLibraries(array $libraries) {
+    $assets = ['css' => [], 'js' => []];
+    foreach ($libraries as $library) {
+      list($extension, $name) = explode('/', $library, 2);
+      $definitions = $this->libraryDiscovery->getLibraryByName($extension, $name);
+      foreach ($definitions as $type => $definition) {
+        // Support css and js assets.
+        if ($type != 'css' && $type != 'js') {
+          continue;
+        }
+        foreach ($definition as $options) {
+          $options += [
+            'type' => 'file',
+            'group' => ($type == 'js') ? JS_DEFAULT : CSS_AGGREGATE_DEFAULT,
+            'weight' => 0,
+            'cache' => TRUE,
+            'preprocess' => TRUE,
+            'attributes' => [],
+            'version' => NULL,
+            'browsers' => [],
+          ];
+
+          // Make sure files get absolute URLs by treating them as external
+          // URLs.
+          if ($options['type'] == 'file') {
+            $default_query_string = $this->state->get('system.css_js_query_string') ?: '0';
+            $query_string = $options['version'] == -1 ? $default_query_string : 'v=' . $options['version'];
+            $query_string_separator = (strpos($options['data'], '?') !== FALSE) ? '&' : '?';
+            $options['data'] = file_create_url($options['data']);
+            $options['data'] .= $query_string_separator . ($options['cache'] ? $query_string : REQUEST_TIME);
+            $options['type'] = 'external';
+          }
+
+          // Always add a tiny value to the weight, to conserve the insertion
+          // order.
+          $options['weight'] += count($definition) / 1000;
+
+          // Local and external files must keep their name as the associative
+          // key so the same JavaScript file is not added twice.
+          $assets[$type][$options['data']] = $options;
+        }
+      }
+    }
+    $css_assets = $this->cssCollectionRenderer->render($assets['css']);
+    $js_assets = $this->jsCollectionRenderer->render($assets['js']);
+    return array_merge($css_assets, $js_assets);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preSave() {
+    // Prepare field build.
+    $field_build = [];
+    $entity = $this->getEntity();
+    $field = $this->getSetting('field');
+    if ($field && isset($entity->{$field}) && $entity->{$field}->getValue()) {
+      $field_build = $entity->{$field}->view('node.full');
+    }
+    $this->value = (string) $this->renderer->renderPlain($field_build);
+    // Prepare assets if there are any.
+    if (!empty($field_build['#attached']['library'])) {
```

Comment:
This logic be problematic on the satellite, let's move this to custom code in hook_node_presave of contentpool_custom_elements